docs/usage.md

```diff
@@ -91,3 +93,143 @@ Find and set the options which fit to your demands, it is also an option to chan
+The software packages are given with their software ID (a number) which can be found on the left side when browsing the software in the `dietpi-software` script. Additionally they are given in the [list of available software IDs](https://github.com/MichaIng/DietPi/wiki/DietPi-Software-list).
+
+As an example the software package "Plex Media Server" has the ID 42.
+
+#### User script
+
+Execution of a user script can be done at these two instants of time:
+
+- pre-networking and pre-DietPi install (`/boot/Automation_Custom_PreScript.sh`)
+- post-networking and post-DietPi install (`/boot/Automation_Custom_Script.sh`)
+
+Search `AUTO_SETUP_CUSTOM_SCRIPT_EXEC` for details.
+
+#### System restore
+
+A restore process can be executed automatically using the keyword AUTO_SETUP_BACKUP_RESTORE. This gives the option to start up a system with a previous made system backup (e.g. for generating many identical systems).
+
+#### General options
+
+These keywords are general ones and need to be used for fully automated setups:
+
+- AUTO_SETUP_ACCEPT_LICENSE
+- AUTO_SETUP_AUTOMATED
+
+### Wait for completion of the automated setup / Logging
+
+The duration of an automated setup depends on the selected options, especially the software packages to be installed and of course of the system's speed. It may take a long time (more than an hour may be possible). To find out whether the automatic installation process is running the user can login to the system via another SSH connection. During the installation process a cyclic message will appear after the login.
+
+The result of the automated installation process is written to the file `/var/tmp/dietpi/logs/dietpi-firstrun-setup.log`.
+
+### Example #1 of file contents
+
+The following lines give an example of needed changes within the `dietpi.txt` file. The target to be achieved is:
+
+- Ethernet connection, DHCP, no proxy, no WiFi
+- German localization
+- Hostname "myDietpiSystem"
+
+Then these entries need to be set within the file `dietpi.txt` prior the initial boot:
+
+```sh
+AUTO_SETUP_ACCEPT_LICENSE=1
+AUTO_SETUP_LOCALE=de_DE.UTF-8
+AUTO_SETUP_KEYBOARD_LAYOUT=de
+AUTO_SETUP_TIMEZONE=Europe/Berlin
+AUTO_SETUP_NET_HOSTNAME=myDietPiSystem
+```
+
+### Example #2 of file contents
+
+The following lines give an example of needed changes within the `dietpi.txt` file. The target to be achieved is:
+
+- Automatic updates
+- Ethernet connection, DHCP, no proxy, no WiFi
+- German localization
+- Hostname "myDietpiSystem"
+- X11 desktop with XFCE and Chromium
+- Autostart to X11 login screen
+
+Then these entries need to be set within the file `dietpi.txt` prior the initial boot:
+
+```sh
+AUTO_SETUP_LOCALE=de_DE.UTF-8
+AUTO_SETUP_KEYBOARD_LAYOUT=de
+AUTO_SETUP_TIMEZONE=Europe/Berlin
+AUTO_SETUP_NET_HOSTNAME=myDietPiSystem
+AUTO_SETUP_DESKTOP_INDEX=-1
+AUTO_SETUP_BROWSER_INDEX=-2
+AUTO_SETUP_AUTOSTART_TARGET_INDEX=16
+AUTO_SETUP_AUTOMATED=1
+AUTO_SETUP_INSTALL_SOFTWARE_ID=25
```

Comment:
Not too important, but when a desktop is installed explicitly, the desktop preference is adjusted accordingly, so it is not necessarily required to change this above. But e.g. if only a VNC server or desktop application is selected via `AUTO_SETUP_INSTALL_SOFTWARE_ID`, then the desktop preference has an effect.

Reply:
Ok, kicked out the AUTO_SETUP_DESKTOP_INDEX line.